packages/wallet-adapter-core/src/new/network.ts

```diff
@@ -0,0 +1,28 @@
+import { Network as AptosNetwork } from '@aptos-labs/ts-sdk';
+
+/**
+ * Subset of `Network` enum that only includes standard networks.
+ */
+export type StandardNetwork = Exclude<AptosNetwork, AptosNetwork.CUSTOM>;
+
+// Note: mapping to the original enum ensures the values are compatible.
+export const StandardNetwork = {
+  MAINNET: AptosNetwork.MAINNET,
+  TESTNET: AptosNetwork.TESTNET,
+  DEVNET: AptosNetwork.DEVNET,
+  LOCAL: AptosNetwork.LOCAL,
+} as const;
+
+/**
+ * Custom network configuration. Requires at least the full node URL to work properly.
+ */
+export interface CustomNetwork {
```

Comment:
this should be aligned with the `AptosConfig` instance in the sdk, right?
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/unit/aptosConfig.test.ts#L71

Reply:
Yes, ideally

I wanted to make the network definition a bit more expressive i.e.
- when using standard networks you just pass the network name as shorthand, everything else will be inferred for you
- when using a custom network, you have to pass it in object form, and some fields are required

at the moment, in the SDK everything other than `network` is optional so you can pass a standard network with overrides, or a custom network with missing fields

if someone is using `mainnet`, `testnet` ... I don't think there's a big use-case for overriding values.
That's what a "custom" network would be. But we can still allow that if needed

Reply:
> when using a custom network, you have to pass it in object form, and some fields are required

Yep, but dont we want to have then pass in an object similar to the one that will eventually be used with the SDK to initiate an Aptos connection?